### PR TITLE
[Frontend] No resource warning

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Context:** PR #656 attempted to remove a warning, but experiments shown that it didn't actually remove the warning.

**Description of the Change:** Revert PR #656. Create derived class from `tempfile.Tempdir` and remove the warning generated in the `_cleanup` method.

**Benefits:** No warning.

**Note**: Needs wheels to test different python versions. I will test once with wheels and then after success I will remove the ci:build-wheels tag. No-cover needs to be added due to different python versions.

### CODE REVIEW

**Comment by dime10** [comment]
You know what still concerns me, is that I don't think the `ResourceWarning` is causing the issue we are seeing in the CI. We should remove that warning for sure, but the logs would indicate there is an *exception* that is raised (during cleanup?) at a point where exceptions are not allowed, thus raising a `UnraisableExceptionWarning`.

**Comment by erick-xanadu** [comment]
> You know what still concerns me, is that I don't think the ResourceWarning is causing the issue we are seeing in the CI. We should remove that warning for sure, but the logs would indicate there is an exception that is raised at a point where exceptions are not allowed (during cleanup?), thus raising an UnraisableExceptionWarning.

I don't think that's the case. I think this is just because the tests say to treat all warnings as an exception. 

(Copy pasted from the other ticket)

```
When this warning is paired with tests that will treat all warnings as errors, such as when using the decorator:

    @pytest.mark.filterwarnings("error")

All warnings will instead raise an exception:

    One can also cause all warnings to be exceptions by using error instead of always.

Which will then make pytest raise another warning

frontend/test/pytest/test_autograph.py::TestForLoops::test_ignore_warnings - pytest.PytestUnraisableExceptionWarning:

which since it is treated as an error, may make a test fail.
```

Otherwise, we would see this exception come up everywhere and not just on tests that treat warnings as errors.

**Comment by dime10** [comment]
> > You know what still concerns me, is that I don't think the ResourceWarning is causing the issue we are seeing in the CI. We should remove that warning for sure, but the logs would indicate there is an exception that is raised at a point where exceptions are not allowed (during cleanup?), thus raising an UnraisableExceptionWarning.
> 
> I don't think that's the case. I think this is just because the tests say to treat all warnings as an exception.
> 
> (Copy pasted from the other ticket)
> 
> ```
> When this warning is paired with tests that will treat all warnings as errors, such as when using the decorator:
> 
>     @pytest.mark.filterwarnings("error")
> 
> All warnings will instead raise an exception:
> 
>     One can also cause all warnings to be exceptions by using error instead of always.
> 
> Which will then make pytest raise another warning
> 
> frontend/test/pytest/test_autograph.py::TestForLoops::test_ignore_warnings - pytest.PytestUnraisableExceptionWarning:
> 
> which since it is treated as an error, may make a test fail.
> ```
> 
> Otherwise, we would see this exception come up everywhere and not just on tests that treat warnings as errors.

Ah that makes sense then. I thought it was raised in other tests as well, and it is, but that is also a no-warning test: `TestJittedWithOneTypeRunWithAnother::test_recompile_no_warning`

File: frontend/catalyst/utils/filesystem.py [hunk 1]
```diff
@@ -53,10 +55,43 @@ def cleanup(self):
...
+    This overrides the _cleanup method which would normally emit a warning
+    after removing the directory. This warning is unconditional and it is emitted
+    because it is not called explicitly.
+    """
+
+    @classmethod
+    def _cleanup(cls, name, warn_message, ignore_errors=False, delete=True):
+        """Ignore ResourceWarning during cleanup"""
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=ResourceWarning)
```
Comment (dime10) [inline frontend/catalyst/utils/filesystem.py:71]: Would it be simpler to just call the [cls._rmtree](https://github.com/python/cpython/blob/0691bd860d3de0c1feb1566d99315a0a582c62a0/Lib/tempfile.py#L935) function directly and skip the warning entirely?

Comment (dime10) [reply frontend/catalyst/utils/filesystem.py:71]: Since you agreed offline I'll just go ahead and resolve the conflict and then merge the PR :)

**Review by dime10 — APPROVED** [review]

**Pull request merged**